src/mongocxx/cursor.cpp

```diff
@@ -14,17 +14,11 @@
 
 #include <mongocxx/cursor.hpp>
 
-#include <cstdint>
-#include <memory>
-#include <string>
-#include <tuple>
-
 #include <bsoncxx/private/libbson.hh>
 #include <bsoncxx/stdx/make_unique.hpp>
 #include <mongocxx/exception/private/mongoc_error.hh>
 #include <mongocxx/exception/query_exception.hpp>
 #include <mongocxx/private/cursor.hh>
-#include <mongocxx/private/libbson.hh>
 #include <mongocxx/private/libmongoc.hh>
 
 #include <mongocxx/config/private/prelude.hh>
```

Comment:
I think I understand the general organization of includes for each file, but I'm just curious on why this particular include is at the very end, separated from other includes?

Reply:
While I'm not sure why this style choice was chosen originally, I'd guess because it's special. Pretty much all of our files need to include the prelude to push important macros. So we wouldn't want to risk this getting removed/modified. Separating it definitely helped _me_ avoid removing it in this commit.

Reply:
CONTRIBUTING.md notes the general layout:
https://github.com/mongodb/mongo-cxx-driver/blob/master/CONTRIBUTING.md#headers
Though it is not enforced by clang format.